### PR TITLE
Update Teleport Connect test plan

### PR DESCRIPTION
**Pull request by ravicious**
TODO:

- [x] Passwordless
- [x] Graceful cert handling
- [x] K8s
- [x] File transfer

[Rendered version](https://github.com/gravitational/teleport/blob/ravicious/update-testplan/.github/ISSUE_TEMPLATE/testplan.md#teleport-connect)

### CODE REVIEW

**Review by gzdunek — APPROVED** [review]

**Review by avatus — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/testplan.md [hunk 1]
```diff
@@ -1352,13 +1377,33 @@ With a default Postgres DB instance, a Teleport instance configured with DB acce
...
+    - For each scenario, create at least one DocumentCluster tab for each available resource kind.
+    - For each scenario, first do the action described in the bullet point, then click "Sync" on the
+      relevant cluster tab. Verify that no unrecoverable error was raised (that is, the app still
+      works). Then restart the app and verify that it was restarted gracefully (no unrecoverable
+      error on restart, the user can continue using the app).
+      - [ ] Stop the root cluster.
+      - [ ] Stop a leaf cluster.
+      - [ ] Disconnect your device from the internet.
+  - DocumentGateway
+    - Create a db connection tab for a given database. Then remove access to that db for that user.
```
Comment (ravicious) [inline .github/ISSUE_TEMPLATE/testplan.md:1390]: Scan the list again and add missing checkboxes.

File: .github/ISSUE_TEMPLATE/testplan.md [hunk 2]
```diff
@@ -1352,13 +1377,33 @@ With a default Postgres DB instance, a Teleport instance configured with DB acce
...
+      Go back to Connect and:
+      - Change database name in the tab. This should still work.
+      - Change database port. This should result in an error.
```
Comment (ravicious) [inline .github/ISSUE_TEMPLATE/testplan.md:1393]: This was written when I was still thinking that restarting a gateway is going to refetch the certs. Restarting a gateway doesn't update certs in any way, so both actions should work.

Look up the test plan for v11 to copy the updated test plan items for this section.

**Comment by zmb3** [comment]
@ravicious can we get this merged in the next week or two so it's ready for the v12 test plan?

**Comment by ravicious** [comment]
> @ravicious can we get this merged in the next week or two so it's ready for the v12 test plan?

Sure; there were always more important things in the pipeline than updating the test plan so I've been postponing this since v11. 🙊

**Review by avatus — APPROVED** [review]

**Review by gzdunek — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**